Marlin/src/module/endstops.cpp

```diff
@@ -554,20 +551,16 @@ void Endstops::update() {
       endstop_poll_count = 7;
       old_live_state = live_state;
     }
-    else if (endstop_poll_count && !--endstop_poll_count)
-      validated_live_state = live_state;
+    else if (endstop_poll_count && --endstop_poll_count) return;
 
-  #else
+    validated_live_state = live_state;
 
-    // Lets accept the new endstop values as valid - We assume hardware filtering of lines
-    esbits_t validated_live_state = live_state;
+    if (!should_check()) return;
 
   #endif
 
-  // Endstop readings are validated in validated_live_state
-
   // Test the current status of an endstop
-  #define TEST_ENDSTOP(ENDSTOP) (TEST(validated_live_state, ENDSTOP))
+  #define TEST_ENDSTOP(ENDSTOP) (TEST(live_state, ENDSTOP))
```

Comment:
this should be validated_live_state, when ENDSTOP_FILTERING is enabled, and 
live_state if ENDSTOP_FILTERING is disabled

Reply:
The value will always be the same at this point in the code.

Reply:
If you keep the return, yes, but that is incorrect. Assume you have already triggered the Z-min limit swithc, and you do a move from Z10 to Z0. 
The Stepper ISR will start execution of that movement, and call endstop.update().
As there was no change of endstop state, then stepper::endstops_triggered(Z_AXIS) will never be called, and the block will completely execute, forcing the endstop...

Reply:
Ah, right. I guess there's still a remote chance that the `validated_state` is still reflecting an old state (triggered) even though the true state is not-triggered, and then the move could be aborted in error. But presumably the chances will be very low now that the noise filter is keeping the state refreshed regardless of `endstops.enabled`.